plugins/repository-hdfs/build.gradle

```diff
@@ -205,10 +258,72 @@ if (secureFixtureSupported) {
   // Fixture dependencies
   integTestSecureCluster.dependsOn secureHdfsFixture, krb5kdcFixture
   integTestSecureRunner.systemProperty 'tests.rest.suite', 'secure_hdfs_repository'
+  integTestSecureRunner.exclude('**/Ha*TestSuiteIT.class') // Ignore HA integration Tests
 } else {
   integTestSecure.enabled = false
 }
 
+// Create a Integration Test suite just for HA related security based tests
+// The following closure must execute before the afterEvaluate block from integTestSecureHA:
+project.afterEvaluate {
+  Path elasticsearchKT = project(':test:fixtures:krb5kdc-fixture').buildDir.toPath().resolve("keytabs").resolve("elasticsearch.keytab").toAbsolutePath()
+  Path krb5conf = project(':test:fixtures:krb5kdc-fixture').buildDir.toPath().resolve("conf").resolve("krb5.conf").toAbsolutePath()
+
+  project.integTestSecureHaCluster.dependsOn(project.bundlePlugin)
+  project.integTestSecureHa.clusterConfig.plugin(project.path)
+  project.integTestSecureHa.clusterConfig.extraConfigFile("repository-hdfs/krb5.keytab", "${elasticsearchKT}")
+
+  final String baseClusterJvmArgs = "-Xms" + System.getProperty('tests.heap.size', '512m') +
+          " " + "-Xmx" + System.getProperty('tests.heap.size', '512m') +
+          " " + "-Djava.security.krb5.conf=${krb5conf}" +
+          " " + System.getProperty('tests.jvm.argline', '')
+  final String jvmArgs
+  if (project.rootProject.ext.javaVersion == JavaVersion.VERSION_1_9) {
+    jvmArgs = baseClusterJvmArgs + " " + '--add-opens=java.security.jgss/sun.security.krb5=ALL-UNNAMED'
+  } else {
+    jvmArgs = baseClusterJvmArgs
+  }
+
+  project.integTestSecureHa.clusterConfig.jvmArgs = jvmArgs
```

Comment:
same comment about code reuse for everything in the afterEvaluate up to this point.